plugins/BEdita/API/src/Model/Action/UpdateAssociatedAction.php

```diff
@@ -165,12 +166,56 @@ function ($a, $b) use ($targetPrimaryKeys) {
                 throw new RecordNotFoundException(__('Record not found in table "{0}"', $type ?: $target->getTable()));
             }
 
-            $meta = Hash::get($datum, '_meta.relation');
-            if (!$this->request->is('delete') && $association instanceof BelongsToMany && $meta !== null) {
-                $targetEntities[$id]->_joinData = $meta;
+            if (!$this->request->is('delete') && $association instanceof BelongsToMany) {
+                $meta = $this->prepareMeta($association, Hash::get($datum, '_meta.relation'));
+                if ($meta !== null) {
+                    $targetEntities[$id]->_joinData = $meta;
+                }
             }
         }
 
         return $targetEntities;
     }
+
+    /**
+     * Prepare relation metadata.
+     *
+     * @param \Cake\ORM\Association&\Cake\ORM\Association\BelongsToMany $association The association.
+     * @param array|null $meta Relation metadata.
+     * @return array|null
+     */
+    protected function prepareMeta($association, $meta)
+    {
+        if (!$association instanceof RelatedTo) {
+            return $meta;
+        }
+
+        $relation = $association->getRelation();
+        if ($relation === null || empty($relation->params)) {
+            return $meta;
+        }
+
+        // Cast stdClass to array recursively
+        $params = json_decode(json_encode($relation->params), true);
+        $defaultParams = array_filter(
+            Hash::get($params, 'properties', []),
+            function (array $prop): bool {
+                return array_key_exists('default', $prop);
+            },
+        );
+        if (count($defaultParams) === 0) {
+            return $meta;
+        }
+        if ($meta === null) {
+            $meta = ['params' => []];
+        }
+
+        foreach ($defaultParams as $name => $data) {
+            if (empty($meta['params'][$name])) {
```

Comment:
I'm really not sure whether the `empty` check is what we really want here. This would result in a field of type `{"type": "boolean", "default": true}` to be essentially always true, cause any attempt to set it as `false` would fall into this branch. Same for an empty string, or `0` as a number.

I'd much rather prefer a `isset` check here, but I'm unsure whether the best thing to do would be to check for key existence altogether using `array_key_exists($meta['params'], $name)`.

Reply:
Assuming we can easily tell whether that parameter is nullable, the (arguably) correct solution would be:

```php
if (!array_key_exists($meta['params'], $name) || (!$paramIsNullable && !isset($meta['params'][$name]))) {
  // Use default
}
```